src/sys/signal.rs

```diff
@@ -619,6 +619,15 @@ impl SigSet {
             Signal::try_from(signum.assume_init()).unwrap()
         })
     }
+
+    /// Converts a `libc::sigset_t` object to a [`SigSet`] without checking  whether the `libc::sigset_t`
+    /// is already initialized. Objects of type`libc::sigset_t` must be initialized by calling either
+    /// [`sigemptyset(3)`](https://man7.org/linux/man-pages/man3/sigemptyset.3p.html) or
+    /// [`sigfillset(3)`](https://man7.org/linux/man-pages/man3/sigfillset.3p.html)
+    /// before being used. In other case, the results are undefined.
+    pub unsafe fn from_sigset_t_unchecked(sigset: libc::sigset_t) -> SigSet {
```

Comment:
It's good practice to have a `# Safety` section for every unsafe pub fun.  Could you please add one?  Also, why does this need to be `unsafe`?  `libc::sigset` doesn't contain any pointers, so it's not obvious what invariant might be violated.

Reply:
> Also, why does this need to be `unsafe`? `libc::sigset` doesn't contain any pointers, so it's not obvious what invariant might be violated.

Good point, it's not related to memory safety or ownership. I marked it as `unsafe`, because In theory it should not be assumed the implementation of signal set, and  the results are undefined if the `sigset_t` it's uninitialized according to posix. 

But, it's also true that to get an uninitialized `libc::sigset_t` it is necessary to use `unsafe`, so perhaps it's better to implement `From`, which is much more convenient than `SigSet::from_sigset_t_unchecked()`. @asomers what do you think, should I change this PR and implement `From` instead?

Reply:
btw, `UContext::sigmask()` and `UContext::sigmask_mut()` both use the pointer cast I mention in the commit message, but I just realize the `SigSet` is not `#[repr(transparent)]`

Reply:
I think leaving it marked as `unsafe` is helpful, as it helps alert the user about a potential footgun in the future regarding ABI compatibility. Here's what POSIX says:

> The implementation of the sigemptyset() (or [sigfillset()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigfillset.html)) function could quite trivially clear (or set) all the bits in the signal set. Alternatively, it would be reasonable to initialize part of the structure, such as a version field, to permit binary-compatibility between releases where the size of the set varies. For such reasons, either sigemptyset() or [sigfillset()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigfillset.html) must be called prior to any other use of the signal set, even if such use is read-only (for example, as an argument to [sigpending()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigpending.html)).

---

> I just realize the SigSet is not #[repr(transparent)]

Can you include this change as part of the PR? I see no reason it can't be marked transparent.

Thanks!

Reply:
If `SigSet` must have the invariant "the inner `sigset` must be initialized with either `sigemptyset` or `sigfillset`", then I suggest eliminating this function entirely, and replacing it with a pair of safe `SigSet::empty() -> Self` and `SigSet::full() -> Self` functions instead.

Reply:
`SigSet::empty` and `SigSet::all` already exist. I believe the intent is to accept a value which was initialized with `sigemptyset` or `sigfillset` and then modified, likely all external to Rust code.

Reply:
> `SigSet::empty` and `SigSet::all` already exist. I believe the intent is to accept a value which was initialized with `sigemptyset` or `sigfillset` and then modified, likely all external to Rust code.

Yes exactly, our use case is in libraries written in rust that will be used from C